aztk/client.py

```diff
@@ -159,13 +178,69 @@ def __get_remote_login_settings(self, pool_id: str, node_id: str):
         result = self.batch_client.compute_node.get_remote_login_settings(pool_id, node_id)
         return models.RemoteLogin(ip_address=result.remote_login_ip_address, port=str(result.remote_login_port))
 
+    def __create_user_on_node(self, username, pool_id, node_id, ssh_key):
+        try:
+            self.__create_user(pool_id=pool_id, node_id=node_id, username=username, ssh_key=ssh_key)
+        except batch_error.BatchErrorException as error:
+            try:
+                self.__delete_user(pool_id, node_id, username)
+                self.__create_user(pool_id=pool_id, node_id=node_id, username=username, ssh_key=ssh_key)
+            except batch_error.BatchErrorException as error:
+                print(error)
+                raise error
+        return ssh_key
+
+    def __create_user_on_pool(self, username, pool_id, nodes):
+        ssh_key = RSA.generate(2048)
+        with concurrent.futures.ThreadPoolExecutor() as executor:
+            futures = {executor.submit(self.__create_user_on_node, username, pool_id, node.id, ssh_key.publickey().exportKey('OpenSSH').decode('utf-8')): node for node in nodes}
+            concurrent.futures.wait(futures)
+        return ssh_key
+
+    def __delete_user_on_pool(self, username, pool_id, nodes):
+        with concurrent.futures.ThreadPoolExecutor() as exector:
+            futures = [exector.submit(self.__delete_user, pool_id, node.id, username) for node in nodes]
+            concurrent.futures.wait(futures)
+
+
+    def __cluster_run(self, cluster_id, container_name, command):
+        pool, nodes = self.__get_pool_details(cluster_id)
+        nodes = [node for node in nodes]
+        cluster_nodes = [self.__get_remote_login_settings(pool.id, node.id) for node in nodes]
+        try:
+            ssh_key = self.__create_user_on_pool('aztk', pool.id, nodes)
+            asyncio.get_event_loop().run_until_complete(ssh_lib.clus_exec_command(command,
+                                                                                  container_name,
+                                                                                  'aztk',
+                                                                                  cluster_nodes,
+                                                                                  ssh_key=ssh_key.exportKey().decode('utf-8')))
+            self.__delete_user_on_pool('aztk', pool.id, nodes)
+        except OSError as exc:
+            raise exc
+
+    def __cluster_copy(self, cluster_id, container_name, source_path, destination_path):
+        pool, nodes = self.__get_pool_details(cluster_id)
+        nodes = [node for node in nodes]
+        cluster_nodes = [self.__get_remote_login_settings(pool.id, node.id) for node in nodes]
+        try:
+            ssh_key = self.__create_user_on_pool('aztk', pool.id, nodes)
```

Comment:
Would it be OK to either use an existing cluster user (if we know it) OR just keep the user around assuming it was created securely?

Reply:
It would be fine to do, but users shouldn't have to care about users to run commands or copy files. 

In general, the entire concept of a user is not really useful to users and can even be confusing. For example, when I ssh in, I specify a user, but regardless of what user is specified, I'm logged into the container as root.